### DISCUSSION TITLE
Serverless Function

### DISCUSSION THREAD

**Discussion by ghost** [vercel/community#1122] [Help] [answered]
Hi !  I'm using a serverless function 

I want to get the current title and year and send it to the handler to make a request.
I'm getting a 404 error.
Do you know where I'm going wrong??

## Home.vue
```js
   const response = await fetch(`/api/get/`,{

          body:JSON.stringify({
            title : title,
            year :year,
          })
        });
```

## handler
```js
export default async function handler(request, response) {
    
    const {title,year} =request.query.body;

    const res = await fetch(`https://www.omdbapi.com/?apikey=${process.env.VUE_APP_API_KEY}&s=${title}&y=${year}`);
    const data = await res.json();
    return response.status(200).json({ data });
}
```

![image](https://user-images.githubusercontent.com/74364667/205955196-a4c09952-ddbd-49f3-8ea4-63ee0faa3345.png)

If you do as below, the title value request is received correctly.
All I want is the title and year.

## Home.vue
```js
 const response = await fetch(`/api/get/${title}`);
```

## handler
```js
export default async function handler(request, response) {

    const {title} =request.query;

    const res = await fetch(`http:s//www.omdbapi.com/?apikey=${process.env.VUE_APP_API_KEY}&s=${title}`);
    const data = await res.json();
    return response.status(200).json({ data });
}

```

![image](https://user-images.githubusercontent.com/74364667/205955552-fc8c67a0-f223-4238-a499-157e0f0c6f53.png)

**Comment by amyegan** [comment, marked as answer]
Good question @Seung-hwan285! You can use multiple [path segments](https://vercel.com/docs/concepts/functions/serverless-functions#path-segments) to pull in more than one value. So my example has `[second].js` nested in `[first]` which allows us to pass two values to the serverless function in `[second].js`. 

<img src="https://user-images.githubusercontent.com/8247105/205971353-636a5ddf-3f7d-42ad-a5a9-fca7a8c29741.png" alt="image of directory nesting multiple path segments" />

Then, you could use something like `example.com/api/red/blue` and get both of the path segment values from `request.query` in the serverless function.

**Reply by ghost** [reply]
@amyegan Thanks  👍  Fixed right away!
